spring-kafka/src/main/java/org/springframework/kafka/listener/adapter/FilteringBatchMessageListenerAdapter.java

```diff
@@ -71,27 +71,31 @@ public FilteringBatchMessageListenerAdapter(BatchMessageListener<K, V> delegate,
 
 		super(delegate, recordFilterStrategy);
 		this.ackDiscarded = ackDiscarded;
+		this.consumerAware = this.delegateType.equals(ListenerType.ACKNOWLEDGING_CONSUMER_AWARE) || this.delegateType.equals(ListenerType.CONSUMER_AWARE);
 	}
 
 	@Override
 	public void onMessage(List<ConsumerRecord<K, V>> records, @Nullable Acknowledgment acknowledgment,
 			Consumer<?, ?> consumer) {
 
-		List<ConsumerRecord<K, V>> consumerRecords = getRecordFilterStrategy().filterBatch(records);
+		final RecordFilterStrategy<K, V> recordFilterStrategy = getRecordFilterStrategy();
+		final List<ConsumerRecord<K, V>> consumerRecords = recordFilterStrategy.filterBatch(records);
 		Assert.state(consumerRecords != null, "filter returned null from filterBatch");
-		boolean consumerAware = this.delegateType.equals(ListenerType.ACKNOWLEDGING_CONSUMER_AWARE)
-						|| this.delegateType.equals(ListenerType.CONSUMER_AWARE);
-		/*
-		 *  An empty list goes to the listener if ackDiscarded is false and the listener can ack
-		 *  either through the acknowledgment
-		 */
-		if (consumerRecords.size() > 0 || consumerAware
-				|| (!this.ackDiscarded && this.delegateType.equals(ListenerType.ACKNOWLEDGING))) {
-			invokeDelegate(consumerRecords, acknowledgment, consumer);
+
+		if (recordFilterStrategy.ignoreEmptyBatch()) {
+			if (acknowledgment != null) {
+				invokeDelegate(consumerRecords, acknowledgment, consumer);
```

Comment:
I don't think this logic is correct. I believe the request was about an option to fully ignore an empty batch as it is right now with a `FilteringMessageListenerAdapter` independently of the manual ack settings.
Therefore this your change is a bit off: looks like you always are going to push an the batch into listener independently of its state. 
My idea with this `ignoreEmptyBatch` was exactly opposite: if it is `true`, we don't call delegate and call `acknowledgment.acknowledge()` if it is present. Otherwise you are right: the container is going to take care about committing offset properly for the whole batch.
We call `invokeDelegate()` in all those cases we talked about and when `ignoreEmptyBatch` is `false`.

Reply:
>  I believe the request was about an option to fully ignore an empty batch as it is right now with a FilteringMessageListenerAdapter independently of the manual ack settings.

I read the request again, and your point is correct. Yes, i make wrong PR. I'm sorry for that.  😓 
I added this logic! 
```java
// I added this.
if (recordFilterStrategy.ignoreEmptyBatch() &&
			consumerRecords.isEmpty() &&
			acknowledgment != null) {
			acknowledgment.acknowledge();
		}
else if (consumerRecords.size() > 0 || this.consumerAware
             || (!this.ackDiscarded && this.delegateType.equals(ListenerType.ACKNOWLEDGING))) {
	     invokeDelegate(consumerRecords, acknowledgment, consumer);
}
...
```
I expect that we are able to meet those requirements with this logic. 
What do you think?

Reply:
Yes, that sounds good. If batch is empty and manual ack, then check for `ignoreEmptyBatch()` and don't call delegate.
I think we all are now on the same page.
👍

Reply:
Yeah, i think so too 👍 
Is there anything else I need to work on? 

from [comment](https://github.com/spring-projects/spring-kafka/issues/2806#issuecomment-2069432409), 
After waiting for the `3.3.x` version to start, the next step for this PR, looks like it will start 🤔.